### ISSUE TITLE
Confirm that full-sync can be forced by maxing out "block_fetch_horizon"

### ISSUE THREAD

**Issue by frol** [near/nearcore#3022]
NEAR Indexer needs a way to get things synced starting from genesis, and @bowenwang1996 [suggested to use a high value to "block_fetch_horizon" (max u64: 18446744073709551615)](https://github.com/nearprotocol/nearcore/pull/2651#discussion_r454701740), so nearcore forces chunks sync instead of just block headers sync.

This needs to be confirmed on NEAR Indexer side as this is what we [recommend tweaking](https://github.com/nearprotocol/nearcore/pull/2997).

**Comment by frol** [comment]
The initial attempt failed when we tried to join an "old" network where there is no single node having the "first" blocks (due to garbage collection). The node with such a config just never got to start syncing.

After the hard-fork, I started the node again and it works fine since then, however, it is not clear if it will do so if I start it now. We need to have another test node with such a config and blank data folder to see if that is going to sync the old data.

@khorolets This still needs further testing.
